core/src/main/java/com/datastrato/gravitino/catalog/Catalog.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog;
+
+import com.datastrato.gravitino.CatalogBasicInfo;
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.file.FilesetCatalog;
+import com.datastrato.gravitino.messaging.TopicCatalog;
+import com.datastrato.gravitino.rel.TableCatalog;
+import com.datastrato.gravitino.schema.SupportsSchemas;
+
+/**
+ * The server-side interface of a catalog. The catalog is the second level entity in the gravitino
+ * system, containing a set of tables.
+ */
+@Evolving
+public interface Catalog extends CatalogBasicInfo {
```

Comment:
I think this should be put into `connector` module, this will be used by developers who develops catalog.

Reply:
ok; moved to the "connector" package in the "core" module.